src/support/generation.c

```diff
@@ -156,25 +156,30 @@ __wt_gen_drain(WT_SESSION_IMPL *session, int which, uint64_t generation)
                 __wt_sleep(0, 10);
 
             /*
-             * If we wait for more than a minute, log the event. In DIAGNOSTIC mode, abort if we
-             * ever wait more than 3 minutes, that's forever.
+             * If we wait for more than a minute, log the event. In diagnostic mode, abort if we
+             * ever wait more than the configured timeout.
              */
             if (minutes == 0) {
                 minutes = 1;
                 __wt_epoch(session, &start);
             } else {
                 __wt_epoch(session, &stop);
                 time_diff_ms = WT_TIMEDIFF_MS(stop, start);
-#define WT_GEN_DRAIN_TIMEOUT_MIN 4
+
                 if (time_diff_ms > minutes * WT_MINUTE * WT_THOUSAND) {
                     __wt_verbose_notice(session, WT_VERB_GENERATION,
                       "%s generation drain waited %" PRIu64 " minutes", __gen_name(which), minutes);
                     ++minutes;
-                    WT_ASSERT(session, minutes < WT_GEN_DRAIN_TIMEOUT_MIN);
                 }
-                /* Enable extra logs 20ms before timing out. */
-                else if (!verbose_timeout_flags &&
-                  time_diff_ms > (WT_GEN_DRAIN_TIMEOUT_MIN * WT_MINUTE * WT_THOUSAND - 20)) {
+
+                /* If no timeout is configured, there is nothing else to do. */
```

Comment:
This comment is slightly confusing, if I understood the description correctly - when no timeout is configured then it would default to 4 minutes, but if we put 0 that means we have configured the timeout it to wait forever.

Reply:
I understand the confusion, thanks for pointing it out and for suggesting something better!

Reply:
is this ok 61a4d23254b18f4d93ab1cf4b5dcd041c60f235e?